client/src/tests/qNa.test.js

```diff
@@ -1,315 +1,103 @@
 import { shallow, mount} from 'enzyme'
 import helper from '../helpers/qnAHelper.js'
 import testData from './QnA-testData'
 import renderer from 'react-test-renderer'
 import QuestionsNAnswers from '../questions-n-answers/qNa'
+import QnAClicks from '../questions-n-answers/QnAClicks.jsx'
 import { expect, jest, test, describe, beforeEach, afterEach } from '@jest/globals'
+
+// import QnAClicks from '../questions-n-answers/QnAClicks.jsx'
 // console.log(testData.test)
 
-const props = {
-  product_id: testData.product_id,
-  data:testData.data,
-  QuestionSavedData: testData.QuestionSavedData,
-  currentItemName: testData.currentItemName,
-  question_id: testData.question_id
-}
 
+// jest.mock('../questions-n-answers/QnAClicks.jsx')
 jest.mock('../clientRoutes/qa.js')
-let wrapper;
-
-describe('QuestionsNAnswers', () => {
-  describe('ComponentDidMount', () => {
-    beforeEach(() => {
-      wrapper = shallow(<QuestionsNAnswers {...props}/>)
-    })
-    afterEach(() => {
-      jest.clearAllMocks()
-    })
-    test('on mount, it executes componentDidMount, filterAnswerNQUestions, showMoreAnsweredQuestions, ShowReportedClass all only once', () => {
-
-
-    const instance = wrapper.instance()
+// jest.mock('../questions-n-answers/qNa')
 
-    console.log(instance)
-    jest.spyOn(instance, 'componentDidMount')
-    instance.componentDidMount()
-    // console.log(instance)
+// jest.mock('../questions-n-answers/QnAClicks.jsx')
 
-    jest.spyOn(instance, 'getReportedAns')
-      return instance.getReportedAns().then(data => {
-        console.log(instance)
-        let answerIds = data.data
 
-        jest.spyOn(instance, 'filterAnswersNQuestions')
-        let sortedData = instance.filterAnswersNQuestions(props.data)
-        // console.log(sortedData[1])
+const props = {
+  product_id: testData.QnAcurrentProductId(),
+  currentItemName: 'camo onsie',
+  data: testData.QnAComponentQuestionsData()
+  // data:testData.data,
+  // QuestionSavedData: testData.QuestionSavedData,
+  // currentItemName: testData.currentItemName,
+  // question_id: testData.question_id
 
-        jest.spyOn(helper, 'showMoreAnsweredQuestions')
-        let showButton = helper.showMoreAnsweredQuestions(sortedData)
+}
 
-        jest.spyOn(helper, 'showReportedClass')
-        let answers = helper.showReportedClass(sortedData[1], answerIds)
+let component;
+describe('QuestionsNAnswers', () => {
+  describe('ComponentDidMount', () => {
+    beforeEach(() => {
 
-        expect(instance.componentDidMount).toBeCalledTimes(1)
-        expect(instance.getReportedAns).toBeCalledTimes(1)
-        expect(instance.filterAnswersNQuestions).toBeCalledTimes(1)
-        expect(helper.showMoreAnsweredQuestions).toBeCalledTimes(1)
-        expect(helper.showReportedClass).toBeCalledTimes(1)
 
-      })
-    })
+     const wrapper = mount(
+        <QnAClicks>
+          {allClicksProps => {
+            return <QuestionsNAnswers allClicksProps={allClicksProps} {...props}/>
```

Comment:
Curious as to how this syntax works. Are allClicksProps merged with other props?